tests/test_frame.py

```diff
@@ -546,6 +546,13 @@ def test_types_groupby() -> None:
     df.groupby(by="col1")
     df.groupby(level="ind")
     df.groupby(by="col1", sort=False, as_index=True)
+
+    cols: list[str] = ["col1", "col2"]
+    df.groupby(by=cols).sum()
+
+    cols_opt: list[str | None] = ["col1", "col2"]
+    df.groupby(by=cols_opt).sum()
```

Comment:
If you want, you can also test the return value

Reply:
Added

Reply:
`check(assert_type(df.groupby(by=cols).sum(), pd.DataFrame), pd.DataFrame)`

Reply:
Wouldn't this also fail with the assignment? Is the assert_type way the proper way of checking this?

Reply:
If the return type was `Any`, `df: pd.DataFrame` would not report an error.

Reply:
Ah, did not think about that. Thanks

Reply:
Independently of this pr: We should check everything like this then, correct?

Reply:
> Independently of this pr: We should check everything like this then, correct?

Ideally yes, but as a starting point, we like new code to use `assert_type` (for static typing) and also `check` (to test the runtime behavior; and for sequence-like items to check the dtpye of the items).